projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/exception.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Any
+from typing import Optional
+
+from vdk.internal.core.errors import BaseVdkError
+from vdk.internal.core.errors import ResolvableBy
+
+
+class IngestionException(BaseVdkError):
+    """
+    Base Exception for all custom exceptions related to the ingestion process.
+    This is intended to catch general exceptions that do not fit into more specific categories.
+    """
+
+    def __init__(self, message: str, vdk_type: ResolvableBy = None):
+        super().__init__(None, vdk_type, message)
+
+
+class IngestionConfigurationException(IngestionException):
```

Comment:
For now, I've kept VDKConfigurationException. But at some point I wondered if I want to have IngestionConfiguraitonException instead. What do you , reviewers think? 

I will remove this exception if we decide to stick to VDKConfigurationException